### ISSUE TITLE
Improve semantics of `hint-mode`'s user options 

### ISSUE THREAD

**Issue by aadcg** [atlas-engineer/nyxt#2865]
We have, in very general terms, two de-facto ways to use `hint-mode`:

1. Use the prompt buffer and match against the hint strings, and other attributes of the hints (the default);
2. Only matches against the hint strings, collapses the prompt buffer and returns as soon as a matching hint string has been typed.

There are roughly 3 user options that enable the 2nd mode of operation above:

1. `nyxt/hint-mode:fit-to-prompt-p`
2.  `nyxt/hint-mode:auto-follow-hints-p`
3. `nyxt/hint-mode:compute-hints-in-view-port-p`

For the 1st option, it must be set to `t` for mode 2 and to `nil` on mode 1.

The 2nd option must be set to `nil` on mode 1 and, arguably, could be set to any boolean on mode 2 (but it would make much more sense to have set to `t`).

The 3rd option, arguably, could be set to any boolean in mode 1 (while it makes much more sense to set it to `nil`). On the other hand, for mode 2 it only makes sense to set to `t` (for performance).

In short, users need to set 3 options to enable mode 2. Also, not all combinations make sense. I think we must do better.

I'm concealing my personal view on a possible solution, so that I don't influence anyone's view.

It'd be great to get some feedback from our users, such as @heiwiper!

**Comment by heiwiper** [comment]
> It'd be great to get some feedback from our users, such as @heiwiper!

I personally prefer if there were two config slots, one for each mode. And since the slots are booleans, I would see it implemented as if we set a default mode slot where it wouldn't matter if it was `nil` or `t` and a higher priority secondary mode where it would override the the current mode if it is set to `t`.

As for the currently available options, I lean more towards your opinion, I don't think the other combination make sense, so it would be better to only use them in the code logic and keep the two config slots I mentioned earlier for the user to modify.

Tha being said, I must say that I am not an experienced Nyxt user, so it would be much appreciated to get feedback from other users.
